common/djangoapps/microsite_configuration/tests/test_backends.py

```diff
@@ -0,0 +1,428 @@
+
+    def has_override_value(self, val_name):
+        """
+        Returns True/False whether a Microsite has a definition for the
+        specified named value
+        """
+        return super(NullBackend, self).has_override_value(val_name)
+
+    def get_all_config(self):
+        """
+        This returns a set of orgs that are considered within all microsites.
+        This can be used, for example, to do filtering
+        """
+        return super(NullBackend, self).get_all_config()
+
+    def get_value_for_org(self, org, val_name, default=None):
+        """
+        This returns a configuration value for a microsite which has an org_filter that matches
+        what is passed in
+        """
+        return super(NullBackend, self).get_value_for_org(org, val_name, default)
+
+    def get_all_orgs(self):
+        """
+        This returns a set of orgs that are considered within a microsite. This can be used,
+        for example, to do filtering
+        """
+        return super(NullBackend, self).get_all_orgs()
+
+    def clear(self):
+        """
+        Clears out any microsite configuration from the current request/thread
+        """
+        return super(NullBackend, self).clear()
+
+
+class BaseMicrositeBackendTests(TestCase):
+    """
+    Go through and test the base abstract class
+    """
+
+    def test_cant_create_instance(self):
+        """
+        We shouldn't be able to create an instance of the base abstract class
+        """
+
+        with self.assertRaises(TypeError):
+            AbstractBaseMicrositeBackend()  # pylint: disable=abstract-class-instantiated
+
+    def test_not_yet_implemented(self):
+        """
+        Make sure all base methods raise a NotImplementedError exception
+        """
+
+        backend = NullBackend()
+
+        with self.assertRaises(NotImplementedError):
+            backend.set_config_by_domain(None)
+
+        with self.assertRaises(NotImplementedError):
+            backend.get_value(None, None)
+
+        with self.assertRaises(NotImplementedError):
+            backend.get_dict(None, None)
+
+        with self.assertRaises(NotImplementedError):
+            backend.is_request_in_microsite()
+
+        with self.assertRaises(NotImplementedError):
+            backend.has_override_value(None)
+
+        with self.assertRaises(NotImplementedError):
+            backend.get_all_config()
+
+        with self.assertRaises(NotImplementedError):
+            backend.clear()
+
+        with self.assertRaises(NotImplementedError):
+            backend.get_value_for_org(None, None, None)
+
+        with self.assertRaises(NotImplementedError):
+            backend.get_all_orgs()
+
+
+@patch(
+    'microsite_configuration.microsite.BACKEND',
+    microsite.get_backend(
+        'microsite_configuration.backends.database.DatabaseMicrositeBackend', SettingsFileMicrositeBackend
+    )
+)
+class DatabaseMicrositeBackendTests(DatabaseMicrositeTest):
```

Comment:
What do you think about moving the database tests out to their own test module to match database.py?  We could do the same thing for the file-based tests.